### ISSUE TITLE
prefer-destructuring maxElements/noBlanks option on array

### ISSUE THREAD

**Issue by mockdeep** [eslint/eslint#8614]
It would be great to have an option to the `prefer-destructuring` rule to only require it when you're taking off from the beginning of the array. Essentially, we like it when it's the first element, or multiple first elements, but having to use commas to go further back in the array we like less:

```js
// good
const [foo] = blah;
const bar = bloo[2];
const [blah, butz] = whatevs;

// bad
const foo = blah.foo;
const [, , bar] = bloo;
```

**Comment by kaicataldo** [comment]
Thanks for contributing! Just so I understand, would turning on this option only allow for array destructuring?

**Comment by mockdeep** [comment]
No, we still want to be able to destructure objects, so this option would be specific to arrays. Maybe `maxArraySkips` or something.

**Comment by mockdeep** [comment]
In our case, we'd like to not have any skips, so I could see it also just being a binary option, either allow skips or not.

**Comment by kaicataldo** [comment]
Thanks for the explanation. My preference would be to do the latter - `disallowArraySkips` (I'm sure there's a better name than that 😄 ) - as I think the intention/problem it is trying to solve is clearer.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
